### PR TITLE
feat: OPX-3.2 opx-base-model

### PR DESCRIPTION
**Pull request by Tejaswi-Goel**
* Update: add FEC supported port speeds
* Update: add supported port speeds for CL91, CL108 and CL74-FC
* Feature: DHCP v4 and v6 Snooping support
* Update: introducing new MAC learning mode 1) os enable, hw disable
        2) os disable, hw enable for Isilon like use-cases where forwarding
        in HW takes place based on ACLs and kernel in the based on bridge MAC learning.
* Update: add the attribute to enable/disable the ARP/Nbr refresh for the HW MAC learning.
* Update: add profile-type attribute in port-group-state and front-panel-port list
* Update: add port-group attribute in the front-panel-port list
* Update: add profile-type attribute in port-group-state and front-panel-port list
* Update: addressed warnings and removed unused attribute.
* Update: admin attribute change to non-mandatory in the trap base model
* Bugfix: revision dates not in descending order in yang model files
* Bugfix: unable-to-apply-icmp-access-list to integration
* Update: add ICMPv6 protocol type
* Update: added support for multicast flood resrict feature
* Update: add trap_grp class for trap grp CPS support
* Update: initial proposal for NAS/NDI trap group object
* Bugfix: add new hashing algo for load balancing traffic
* Feature: initial checkin for openconfig data model
* Update: Copyright year
* Update: Upgrading major version due to introduction of backward incompatible changes

Signed-off-by: Tejaswi Goel <Tejaswi_Goel@Dell.com>

### CODE REVIEW

**Comment by Tejaswi-Goel** [comment]
Pull-request updated, HEAD is now 7e89cc4d8556b641a1547290c6835700ef9c64f1

File: yang-models/dell-base-cleanup-events.yang [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    organization "Dell EMC";
+    contact
+        "http://www.dell.com/support";
+    description
+        "This module contains a collection of common YANG derived RPC(Remote Procedure Call).
+        This can be augmented in other Base Yang modules to synchronously cleanup the routes
+        and other resources.
+
+        Copyright (c) 2015-2018 by Dell EMC, All rights reserved.";
```
Comment (rakeshdatta) [inline yang-models/dell-base-cleanup-events.yang]: Change the copyright

File: yang-models/l3-multicast.yang [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+    organization "Dell EMC";
+    contact "http://www.dell.com/support";
+
+    description "This module contains collection of YANG definitions for
+                 managing layer 3 multicast routing. Using this model multicast applications
+                 can enable Layer 3 multicast on VRF and per interface basis, configure and get
+                 multicast route and it's status.
+
+                 Copyright (c) 2015-2018 by Dell EMC, All rights reserved.";
```
Comment (rakeshdatta) [inline yang-models/l3-multicast.yang]: copyright!

**Review by rakeshdatta — CHANGES_REQUESTED** [review]

**Comment by Tejaswi-Goel** [comment]
Pull-request updated, HEAD is now ac810954cda2d1380efa7d4d167b22dfa2ff6aca

**Comment by Tejaswi-Goel** [comment]
Pull-request updated, HEAD is now 2758218e3afbcce22a15139a57d37a48b2bbd643

**Review by hpersh — APPROVED** [review]

**Review by rakeshdatta — APPROVED** [review]

**Pull request merged**